server/services/userService.test.ts

```diff
@@ -52,4 +54,42 @@ describe('User service', () => {
       await expect(userService.getActingUser(callConfig)).rejects.toEqual(new Error('some error'))
     })
   })
+
+  describe('getActingUserPrimaryNavigationList', () => {
+    const communityAccommodationUser = userFactory.build()
+
+    it('returns an empty list by default', () => {
+      expect(userService.getActingUserPrimaryNavigationList(communityAccommodationUser, '')).toEqual([])
+    })
+
+    describe('when user is an assessor', () => {
+      const assessorUser = userFactory.build({ roles: ['assessor'] })
+      const navigationList: Array<PrimaryNavigationItem> = [
+        { text: 'Bookings', href: paths.bookings.index({}) },
+        { text: 'Manage properties', href: paths.premises.index({}) },
+        { text: 'Referrals', href: paths.assessments.index({}) },
+        { text: 'Search bedspaces', href: paths.bedspaces.search({}) },
+        { text: 'Reports', href: paths.reports.index({}) },
+      ]
+
+      it('returns a list of navigation items for the user', () => {
+        const result = userService.getActingUserPrimaryNavigationList(assessorUser, '')
+        expect(result).toEqual(navigationList)
+      })
+
+      describe('when a user is viewing a nav list item page', () => {
+        it.each(navigationList)(`returns active when a user is viewing %p page with %p link`, ({ text, href }) => {
```

Comment:
Could this also test the correct behaviour with paths that only _contain_ the nav url? I.e. showing 'Bookings' as active for `/bookings/foo/bar`